File: examples/01_simple.py
"""Crawlingo: Basic page fetch with CSS and XPath selectors.

Usage:
    pip install crawlingo
    python 01_simple.py
"""

from crawlingo import Page


def main():
    print("=== Crawlingo Simple Extraction Example ===")

    url = "https://httpbin.org/html"
    print(f"Fetching {url}...")

    page = Page(url)

    print(f"\nResponse Status: {page.status}")
    print(f"Page Title: '{page.title()}'")

    print("\n--- CSS Selector (h1) ---")
    h1 = page.css("h1")
    print(f"Found {len(h1)} h1 elements. Text: '{h1.text()}'")

    print("\n--- XPath Selector (//p) ---")
    paragraphs = page.xpath("//p")
    for i, p in enumerate(paragraphs):
        print(f"Paragraph {i+1}: '{p.text()}'")

    print("\n--- Text Anchors ---")
    melville_el = page.find_text("Herman Melville")
    print(f"Found 'Herman Melville': {len(melville_el) > 0}")
    if melville_el:
        print(f"Outer HTML: {melville_el.first().html()}")

    print("\n--- Clean Markdown ---")
    print(page.markdown()[:300])


if __name__ == "__main__":
    main()

File: examples/01_simple.js
const { Page } = require('../sdk/nodejs');

async function main() {
  console.log("=== Crawlingo Node.js Simple Extraction Example ===");

  const url = "https://httpbin.org/html";
  console.log(`Fetching ${url}...`);

  try {
    const page = await Page.create(url);

    console.log(`\nResponse Status: ${page.status}`);
    console.log(`Page Title: '${page.title()}'`);

    // CSS Selector
    console.log("\n--- CSS Selector (h1) ---");
    const h1 = page.css("h1");
    console.log(`Found ${h1.length} h1 elements. Text: '${h1.text.join(", ")}'`);

    // XPath Selector
    console.log("\n--- XPath Selector (//p) ---");
    const paragraphs = page.xpath("//p");
    for (let i = 0; i < paragraphs.length; i++) {
      console.log(`Paragraph ${i + 1}: '${paragraphs.at(i).text}'`);
    }

    // Text Anchor
    console.log("\n--- Text Anchors ---");
    const melvilleEl = page.findText("Herman Melville");
    console.log(`Found 'Herman Melville': ${melvilleEl.length > 0}`);
    if (melvilleEl.length > 0) {
      console.log(`Outer HTML: ${melvilleEl.first().html}`);
    }
  } catch (error) {
    console.error("An error occurred during page fetch:", error);
  }
}

main();
